### ISSUE TITLE
Postgres Dialect: Support for the COPY statement

### ISSUE THREAD

**Issue by derickl** [sqlfluff/sqlfluff#2480]
## Expected Behaviour
I would expect that `COPY...TO` and `COPY...FROM` statements are supported by SqlFluff as they are supported by Postgres ([docs](https://www.postgresql.org/docs/13/sql-copy.html))

## Observed Behaviour
```shell
sqlfluff lint --dialect postgres test.sql
== [sample.sql] FAIL                                                                                                   
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'COPY
                       | (\nSelect my_tablecol From my_table\n...'
```

```shell
sqlfluff parse --dialect postgres test.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'StatementSegment'
[L:  1, P:  1]      |            raw:                                              'COPY'
[L:  1, P:  5]      |            whitespace:                                       ' '
[L:  1, P:  6]      |            bracketed:
[L:  1, P:  6]      |                start_bracket:                                '('
[L:  1, P:  7]      |                newline:                                      '\n'
[L:  2, P:  1]      |                raw:                                          'Select'
[L:  2, P:  7]      |                whitespace:                                   ' '
[L:  2, P:  8]      |                raw:                                          'my_tablecol'
[L:  2, P: 19]      |                whitespace:                                   ' '
[L:  2, P: 20]      |                raw:                                          'From'
[L:  2, P: 24]      |                whitespace:                                   ' '
[L:  2, P: 25]      |                raw:                                          'my_table'
[L:  2, P: 33]      |                newline:                                      '\n'
[L:  3, P:  1]      |                end_bracket:                                  ')'
[L:  3, P:  2]      |            whitespace:                                       ' '
[L:  3, P:  3]      |            raw:                                              'to'
[L:  3, P:  5]      |            whitespace:                                       ' '
[L:  3, P:  6]      |            raw:                                              "'/tmp/dump.csv'"
[L:  3, P: 21]      |            whitespace:                                       ' '
[L:  3, P: 22]      |            bracketed:
[L:  3, P: 22]      |                start_bracket:                                '('
[L:  3, P: 23]      |                raw:                                          'FORMAT'
[L:  3, P: 29]      |                whitespace:                                   ' '
[L:  3, P: 30]      |                raw:                                          'csv'
[L:  3, P: 33]      |                comma:                                        ','
[L:  3, P: 34]      |                whitespace:                                   ' '
[L:  3, P: 35]      |                raw:                                          'HEADER'
[L:  3, P: 41]      |                comma:                                        ','
[L:  3, P: 42]      |                whitespace:                                   ' '
[L:  3, P: 43]      |                raw:                                          'DELIMITER'
[L:  3, P: 52]      |                whitespace:                                   ' '
[L:  3, P: 53]      |                raw:                                          "'#'"
[L:  3, P: 56]      |                end_bracket:                                  ')'
[L:  3, P: 57]      |    statement_terminator:                                     ';'
[L:  3, P: 58]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'COPY (\nSelect
                       | my_tablecol From my_table\n...'
```
## Steps to Reproduce
Create `test.sql` with the following query
```sql
COPY (
Select my_col From my_table
) TO '/tmp/dump.csv' WITH (FORMAT csv, HEADER, DELIMITER '#');
```

## Dialect
Postgres

## Version
sqlfluff 0.9.2
python 3.8.12

## Configuration
Default
